tensorboard/java/org/tensorflow/tensorboard/vulcanize/Vulcanize.java

```diff
@@ -321,7 +321,14 @@ private static Node visitScript(Node node) throws IOException {
       script = INLINE_SOURCE_MAP_PATTERN.matcher(script).replaceAll("");
     }
     boolean wantsMinify = getAttrTransitive(node, "jscomp-minify").isPresent();
-    if (node.attr("src").endsWith(".min.js")
+
+    if (node.hasAttr("jscomp-externs")) {
+      Path filePath = getWebfile(path);
+      SourceFile sourceFile = SourceFile.fromCode(path.toString(), script);
+      externs.add(sourceFile);
```

Comment:
Is duplication an issue? If so, `externs` can be a `LinkedHashSet`.

Reply:
Added `LinkedHashMap` to dedup based on the file name. As I alluded to in the commit message, not all extern files have webpath and, hence, I decided to dedup with the file path.